kong/plugins/jwt/jwt_parser.lua

```diff
@@ -282,6 +282,25 @@ function _M:verify_registered_claims(claims_to_verify)
   return errors == nil, errors
 end
 
+--- check the maximum expiration 
+-- @param maximum_expiration of the claim, or -1 if no expiration
+function _M:check_maximum_expiration(maximum_expiration)
+
+  if maximum_expiration <= 0 then
+     return true, nil
+  end
+  local exp = self.claims["exp"]
+  if exp then
+    if (exp - ngx_time()) > maximum_expiration then
+      return false, {exp = "exceeds maximum expiration"}
+    end
+  else
+      return false, {exp = "is missing"}
+  end
```

Comment:
What do you think about: "exceeds maximum _allowed_ expiration"?

Reply:
Also, as per our contributing guide, we recommend hoisting short branches with early exits for clarity, and a few other minor nitpicks:
```lua
local exp = self.claims.exp
if not exp then
  return false, { exp = "missing" }
end

if exp - ngx_time() > maximum_expiration then
  return false, { exp = "exceeds maximum allowed expiration" }
end

return true
```

Reply:
I changed it to the message "exceeds maximum allowed expiration"  if the 'exp' is missing.